core/src/main/java/org/teavm/backend/wasm/generate/WasmDependencyListener.java

```diff
@@ -55,6 +68,20 @@ public void transformClass(ClassHolder cls, ClassReaderSource innerSource, Diagn
                 method.setProgram(null);
                 method.getModifiers().add(ElementModifier.NATIVE);
             }
+
+            AnnotationReader exportAnnot = method.getAnnotations().get(Export.class.getName());
+            if (exportAnnot != null) {
+                exportedMethods.add(method);
+
+            }
+        }
+    }
+
+    @Override
+    public void beforeCompleting(DependencyAgent agent) {
```

Comment:
Well, methodReached is not called for @Export annotated methods. transformClass checks for @Export methods and adds them to dependency analysis by caching them. beforeCompleting is a new lifecycle method i introduced to allow additional stuff like this *after* dependency analysis was completed and *before* completion stage